src/envs/base_env.py

```diff
@@ -81,3 +115,24 @@ def seed(self, seed: int) -> None:
         self._train_rng = np.random.default_rng(self._seed)
         self._test_rng = np.random.default_rng(self._seed +
                                                CFG.test_env_seed_offset)
+
+    def reset(self, train_or_test: str, task_idx: int) -> State:
+        """Resets the current state to the train or test task initial state."""
+        task = self.get_task(train_or_test, task_idx)
+        self._current_state = task.init
```

Comment:
should we also do task.init.copy() here? I know that Tasks are immutable, but didn't we see before that we could inadvertently change attributes, e.g. if we do `task.init.set(...)`. OTOH, we use this pattern in a lot of places without making copies, so maybe fine as is?

Reply:
either way is okay with me

Reply:
ignoring. we should be careful to always copy states before doing any `.set` stuff, which we already are in all our simulate functions